### PR TITLE
Enable self transition

### PR DESCRIPTION
**Pull request by bmionescu**
This change causes self-transitions to run state callbacks for the state self-transitioned to, instead of ignoring the callback. The current looplab/fsm library, from which this was forked, checks for self-transition when running an Event and returns early, before running any state callbacks. We simply move that check to after state callbacks are run. The self-transition error is still returned, it just happens after we run the callback of the state that was self-transitioned to.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,3 +1,3 @@
-module github.com/looplab/fsm
+module github.com/MonochromeHousing/fsm
 
 go 1.16
```
Comment (dgfiloso) [inline go.mod]: I think it would be good to upgrade to latest version which is 1.22.5, and confirm that there aren't any problems

Comment (bmionescu) [reply go.mod]: You mean go version (not fsm version I assume) ?

Comment (bmionescu) [reply go.mod]: Done

File: fsm.go [hunk 2]
```diff
@@ -396,6 +387,18 @@ func (f *FSM) Event(ctx context.Context, event string, args ...interface{}) erro
...
+                // f.afterEventCallbacks needn't be called when the if block is moved here,
+                // because it is called in transitionFunc
+                //f.stateMu.RUnlock()
+                //defer f.stateMu.RLock()
+                //f.eventMu.Unlock()
+                //unlocked = true
+                //f.afterEventCallbacks(ctx, e)
+                return NoTransitionError{e.Err}
+        }
+
```
Comment (dgfiloso) [inline fsm.go]: This part is not really necessary, it can be deleted

**Review by dgfiloso — APPROVED** [review]

**Pull request merged**